File: algorithms/python3/binary_search.py
def binary_search(L, e):
    """
    List Element -> Integer
    Given a list and an element,
    return the index of the element in the list, or nil if not found
    """
    def helper(lo, hi):
        if lo > hi:
            return False
        mid = (lo + hi) // 2
        if L[mid] == e:
            return mid
        elif L[mid] > e:
            return helper(lo, mid-1)
        else:
            return helper(mid+1, hi)
    return helper(0, len(L)-1)

File: algorithms/clojure/binary_search.clj
(defn binary-search
  "Vector Element -> Integer, Given a vector and an element, return the index of the element in the vector, or nil if not found"
  [v e]
  (loop [lo 0
         hi (dec (count v))]
    (if (> lo hi) nil
        (let [md (quot (+ lo hi) 2)
              mv (v md)]
          (cond
            (> mv e) (recur lo (dec md))
            (< mv e) (recur (inc md) hi)
            (= mv e) md)))))
